### ISSUE TITLE
gix clone pytorch panics

### ISSUE THREAD

**Issue by 0xdeafbeef** [Byron/gitoxide#594]
### Duplicates

- [X] I have searched the existing issues

### Current behavior 😯

```
thread '<unnamed>' panicked at 'not yet implemented', /home/odm3n/.cargo/registry/src/github.com-1ecc6299db9ec823/git-worktree-0.7.0/src/index/entry.rs:101:43 ===   ===   ===   ===   ===   ===   ===   ===   ===]
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at 'not yet implemented', /home/odm3n/.cargo/registry/src/github.com-1ecc6299db9ec823/git-worktree-0.7.0/src/index/entry.rs:101:43
thread 'main' panicked at 'no panic: Any { .. }', /home/odm3n/.cargo/registry/src/github.com-1ecc6299db9ec823/git-features-0.23.1/src/parallel/in_parallel.rs:84:6   ===   ===   ===   ===   ===   ===   ===   ===]
```

### Steps to reproduce 🕹

1. `gix -v clone https://github.com/pytorch/pytorch.git pytorch`
```
gix --version
gix-plumbing 0.17.0
```

**Comment by Byron** [comment]
Thanks for reporting, I am looking into it!

I tried with a slightly newer version and didn't get a panic, but a failed clone nonetheless:

```
❯ RUST_BACKTRACE=full gix -v clone https://github.com/pytorch/pytorch.git pytorch
 10:16:31 read pack done 995.1MB in 547.60s (1.8MB/s)
 10:16:31  indexing 1839478 objects were resolved into 919739 objects during thin-pack resolution
 10:16:31  indexing done 919.7k objects in 542.64s (1.7k objects/s)
 10:16:31 decompressing done 2.1GB in 542.64s (3.9MB/s)
 10:16:33     Resolving done 919.7k objects in 2.11s (436.9k objects/s)
 10:16:33      Decoding done 16.9GB in 2.11s (8.0GB/s)
 10:16:33 writing index file done 25.8MB in 0.03s (740.7MB/s)
 10:16:33  create index file done 919.7k objects in 544.91s (1.7k objects/s)
Error: Failed to update references to their new position to match their remote locations

Caused by:
    0: A lock could not be obtained for reference "refs/remotes/origin/update-test-distribute"
    1: The lock for resource 'pytorch/.git/refs/remotes/origin/update-test-distribute could not be obtained after 0.10s after 8 attempt(s). The lockfile at 'pytorch/.git/refs/remotes/origin/update-test-distribute.lock' might need manual deletion.
```

I am investigating what kind of refs could cause that.

**Comment by Byron** [comment]
Thanks for providing such a nice test-repository, I will keep it and maybe remember to post here once this is working. The error I ran into related to the lock being taken can be reproduced as well, I opened a separate issue for that (#595).

There reason I am closing this issue as now (that I can reproduce it) it's clear that `not yet implemented` is intended and this should work towards the end of this year or first quarter of next year.

Thanks for your understanding.
